core/services/bulletprooftxmanager/eth_confirmer.go

```diff
@@ -723,34 +731,64 @@ func (ec *EthConfirmer) handleAnyInProgressAttempts(ctx context.Context, address
 	return nil
 }
 
-func getInProgressEthTxAttempts(ctx context.Context, db *gorm.DB, address gethCommon.Address, chainID big.Int) ([]EthTxAttempt, error) {
-	var attempts []EthTxAttempt
-	err := db.
-		WithContext(ctx).
-		Preload("EthTx").
-		Joins("INNER JOIN eth_txes ON eth_txes.id = eth_tx_attempts.eth_tx_id AND eth_txes.state in ('confirmed', 'confirmed_missing_receipt', 'unconfirmed')").
-		Where("eth_tx_attempts.state = 'in_progress' AND eth_txes.from_address = ? AND eth_txes.evm_chain_id = ?", address, chainID.String()).
-		Find(&attempts).Error
+func getInProgressEthTxAttempts(ctx context.Context, db *sqlx.DB, lggr logger.Logger, address gethCommon.Address, chainID big.Int) (attempts []EthTxAttempt, err error) {
+	err = postgres.NewQ(db, postgres.WithParentCtx(ctx)).Transaction(lggr, func(q postgres.Queryer) error {
+		err = q.Select(&attempts, `
+SELECT eth_tx_attempts.* FROM eth_tx_attempts
+INNER JOIN eth_txes ON eth_txes.id = eth_tx_attempts.eth_tx_id AND eth_txes.state in ('confirmed', 'confirmed_missing_receipt', 'unconfirmed')
+WHERE eth_tx_attempts.state = 'in_progress' AND eth_txes.from_address = $1 AND eth_txes.evm_chain_id = $2
+`, address, chainID.String())
+		if err != nil {
+			return errors.Wrap(err, "getInProgressEthTxAttempts failed to load eth_tx_attempts")
+		}
+		err = loadEthTxes(q, attempts)
+		return errors.Wrap(err, "getInProgressEthTxAttempts failed to load eth_txes")
+	})
 	return attempts, errors.Wrap(err, "getInProgressEthTxAttempts failed")
 }
 
+func loadEthTxes(q postgres.Queryer, attempts []EthTxAttempt) error {
+	ethTxM := make(map[int64]EthTx)
+	for _, attempt := range attempts {
+		ethTxM[attempt.EthTxID] = EthTx{}
+	}
+	ethTxIDs := make([]int64, len(ethTxM))
+	var i int
+	for id := range ethTxM {
+		ethTxIDs[i] = id
+		i++
+	}
```

Comment:
Is this meant to collect the `EthTxAttempt.ID`s?
```suggestion
	for i, txa := range ethTxM {
		ethTxIDs[i] = txa.ID
	}
```

Reply:
Yes but I don't think your suggestion will work. Since we are iterating over a map, that `i` is not an index - it's the key which is the id.

Reply:
Oh, I totally misread. What about constructing both at once in the first loop?